### PR TITLE
PlayerAttemptPickupItemEvent additions

### PR DESCRIPTION
**Pull request by BillyGalbreath**
Replaces #753 

I finally figured out how to edit existing patches. ;)

### CODE REVIEW

File: Spigot-API-Patches/0055-PlayerAttemptPickupItemEvent.patch [hunk 1]
```diff
@@ -37,6 +42,34 @@ index 00000000..2f5fdb13
...
++    public void setFlyAtPlayer(boolean flyAtPlayer) {
++        this.flyAtPlayer = flyAtPlayer;
++    }
++
++    /**
++     * Gets if the item will fly at the player
++     *
++     * @return True if the item will fly at the player
++     */
++    public boolean getFlyAtPlayer() {
```
Comment (potatostealer) [inline Spigot-API-Patches/0055-PlayerAttemptPickupItemEvent.patch:69]: getFlyAtPlayer is misleading, granted it gives the implication that you're returning an object that WILL fly at the player.
if possible, willFlyAtPlayer would be better imo

File: Spigot-Server-Patches/0215-PlayerAttemptPickupItemEvent.patch [hunk 2]
```diff
@@ -16,19 +16,30 @@ index fed1f4ab2..a580a5866 100644
...
-@@ -329,6 +330,12 @@ public class EntityItem extends Entity implements HopperPusher {
-             Item item = itemstack.getItem();
-             int i = itemstack.getCount();
- 
+@@ -332,7 +333,22 @@ public class EntityItem extends Entity implements HopperPusher {
+             // CraftBukkit start - fire PlayerPickupItemEvent
+             int canHold = entityhuman.inventory.canHold(itemstack);
+             int remaining = i - canHold;
+-            boolean flyAtPlayer = false; // Paper
++
```
Comment (potatostealer) [inline Spigot-Server-Patches/0215-PlayerAttemptPickupItemEvent.patch:24]: unnecessary empty space?

**Review by HuManKeat — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0215-PlayerAttemptPickupItemEvent.patch (hunk 2 above)
Comment (BillyGalbreath) [reply Spigot-Server-Patches/0215-PlayerAttemptPickupItemEvent.patch:24]: Empty space isn't mine. It's just the diff is weird (it removes empty space but adds it back?)

File: Spigot-API-Patches/0055-PlayerAttemptPickupItemEvent.patch (hunk 1 above)
Comment (BillyGalbreath) [reply Spigot-API-Patches/0055-PlayerAttemptPickupItemEvent.patch:69]: Should have said something last month with PlayerPickuoItemEvent addition. Renaming here now would make it inconsistent.

**Comment by zachbr** [comment]
d7039347b06ecc8c65e3dbe0ca532d83960c75ef
Thanks

**Pull request closed without merging**